quickwit-search/src/client.rs

```diff
@@ -19,39 +19,132 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+use std::fmt;
 use std::net::SocketAddr;
+use std::sync::Arc;
 
 use http::Uri;
 use tonic::transport::Channel;
 use tonic::transport::Endpoint;
+use tonic::Request;
 
-use quickwit_proto::search_service_client::SearchServiceClient;
+use crate::error::parse_grpc_error;
+use crate::SearchError;
+use crate::SearchService;
 
-#[derive(Debug, Clone)]
-pub struct WrappedSearchServiceClient {
-    client: SearchServiceClient<Channel>,
+/// Impl is an enumlation that meant to manage Quickwit's search service client types.
+#[derive(Clone)]
+enum Impl {
```

Comment:
Impl is a too generic name

Reply:
@PSeitz 
How about the name `SearchServiceClientImpl`?

Reply:
yes that's better